src/rust/rust-proto-new/src/graplinc/grapl/pipeline/v1beta1.rs

```diff
@@ -0,0 +1,236 @@
+use std::time::SystemTimeError;
+
+use bytes::{
+    Buf,
+    BufMut,
+    Bytes,
+};
+use prost::Message;
+
+use crate::{
+    graplinc::common::v1beta1::{
+        SystemTime,
+        Uuid,
+    },
+    protobufs::graplinc::grapl::pipeline::v1beta1::{
+        Envelope as _Envelope,
+        Metadata as _Metadata,
+        RawLog as _RawLog,
+    },
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+//
+// Metadata
+//
+
+#[derive(Debug)]
+pub struct Metadata {
+    tenant_id: Uuid,
+    trace_id: Uuid,
+    retry_count: u32,
+    created_time: SystemTime,
+    last_updated_time: SystemTime,
+    event_source_id: Uuid,
+}
+
+impl TryFrom<_Metadata> for Metadata {
+    type Error = SerDeError;
+
+    fn try_from(metadata_proto: _Metadata) -> Result<Self, Self::Error> {
+        let tenant_id = if let Some(tenant_id) = metadata_proto.tenant_id {
+            tenant_id
+        } else {
+            return Err(SerDeError::MissingField("tenant_id absent".to_string()));
```

Comment:
I'd remove the "absent" bit from all of these. The string should just be the field name. Something like "absent" belongs in the Error's Display or Debug impls.

Reply:
fixed